### PR TITLE
[iOS] Add Clipboard debug info to help with Lost Tabs debugging.

### PR DESCRIPTION
**Pull request by iccub**
Resolves https://github.com/brave/brave-browser/issues/38449

Security review https://github.com/brave/reviews/issues/1624

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: ios/brave-ios/Sources/Brave/Utility/AppDebugComposer.swift [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        }
+      }
+
+      return result
+    }
+
+    let result =
+      """
+      \(printDeviceInfo)
+      \(printFolderTreeStructure)
```
Comment (stoletheminerals) [inline ios/brave-ios/Sources/Brave/Utility/AppDebugComposer.swift:73]: Will tree structure contain `Documents/Downloads` folder? We don't want users to share this with us. We should at least skip that folder or obfuscate file names/extensions.

Comment (iccub) [reply ios/brave-ios/Sources/Brave/Utility/AppDebugComposer.swift:73]: This has not changed. I renamed the file so this class now handles both app structure and the Tab debug info.
Github shows it like this unfortunately
This is not called in the new code I added

File: ios/brave-ios/Sources/Brave/Utility/AppDebugComposer.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    let model = device.userInterfaceIdiom == .pad ? "iPad" : "iPhone"
+    let iOSVersion = "\(device.systemName) \(device.systemVersion)"
+    let appVersion =
+      Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String ?? "-"
+
+    return "Setup: \(model), \(iOSVersion), Brave \(appVersion)"
+  }
+
+  /// This function prepares data to help us identify any app storage problems users may have.
+  static func composeAppSize() -> String {
```
Comment (iccub) [inline ios/brave-ios/Sources/Brave/Utility/AppDebugComposer.swift:25]: This method is unchanged, renamed so this class can handle more copy-clipboard actions

**Review by Brandon-T — APPROVED** [review]

**Review by soner-yuksel — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **PASSED** on `iPhone 11` running `iOS 17.5.1` using the following build(s):

```
1.68.51 Chromium: 126.0.6478.17 (Official Build) nightly (64-bit) 
--- | ---
Revision | e6fb0e27c36d...
OS | iOS
```

Using the STR/Cases outlined via https://github.com/brave/reviews/issues/1624#issuecomment-2125535694, ensured that tappin on `Copy Tabs info to clipboard` dumps the following data into the clipboard:

```
Setup: iPhone, iOS 17.5.1, Brave 1.68
Managed tabs: 1
Managed regular tabs: 1
Managed private tabs: 0
Storage Tabs: 1
Storage Regular Tabs: 1
Storage Private Tabs: 0
Storage TabGroup: 0
Storage Window: 1
Recently Closed tabs: 0
Tab sync enabled: false
Window IDs of Tabs: ["793820EF"]
Unique Window IDs: ["793820EF"]
Tabs without window: 0
Tabs without group: 1
```

IMG_0253

**Pull request merged**